src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exeptions.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.models.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final Logger logger = LogManager.getLogger(ManufacturerDaoImpl.class);
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        logger.info("Method create was called with manufacturer: " + manufacturer);
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerRequest,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t insert manufacturer to db " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        logger.info("Method get was called with id: " + id);
+        Manufacturer manufacturer = new Manufacturer();;
+        String getManufacturersRequest = "SELECT * FROM manufacturers WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturer =
+                        connection.prepareStatement(getManufacturersRequest)) {
+            getManufacturer.setObject(1, id);
+            ResultSet resultSet = getManufacturer.executeQuery();
+            if (resultSet.next()) {
+                String manufacturerName = resultSet.getString("name");
+                String manufacturerCountry = resultSet.getString("country");
+                Long idManufacturer = resultSet.getObject("id", Long.class);
+                manufacturer.setName(manufacturerName);
+                manufacturer.setCountry(manufacturerCountry);
+                manufacturer.setId(idManufacturer);
+            }
+            return Optional.of(manufacturer);
```

Comment:
```suggestion
            return Optional.of(manufacturer);
```

What's the point of Optional if it will always be with Manufacturer? We do not check in any way whether it is empty or not.